File: Programmers/JeongHyun/4주차 (정렬)/가장 큰 수.js
function solution(numbers) {
    var answer ='';
  //(b+a) -(a+b) 
  // 6,10 이 있는 경우 '610' , '106' 중 더 큰 수 비교해주기 위함
     function compareNumbers(a,b){
        return (b+a)-(a+b);
    }
  /*
  [6,10,2] => join => 6,10,2 (string) => split(',') => ['6','10','2'] (object) => sort 진행
  string 6,10,2 에서는 sort가 안되기 때문에 각각의 string으로 바꾸어서 진행하였습니다.
  */  
    answer = numbers.join().split(',').sort(compareNumbers).join('');
  // testcase 11에서 실패가 떠서 질문하기에서 예제를 보니 [0,0,0,0] 인 경우 '0'을 return 해줘야하는 규칙을 넣어줬습니다.
    return answer[0]==='0' ? '0' :answer;
}

File: Programmers/JiHo/4주차(정렬)/가장 큰 수.py
'''
python sort lambda : list로 리턴시 list요소를 기준으로 정렬
lambda x:list(str(x))으로 실행시
-> 34, 30, 3으로 정렬으로 2번예제를 충족하지 못함
-> list(str(x)) * 3으로 비교할수있는 최저 원소를 채워준다(원소최대 1000)
즉, 3 -> 333, 34 -> 343434, 30 -> 303030으로 sort시 list의 앞부분부터 비교되어 정렬되기때문에
두번째, 세번째요소까지 비교하여 정렬가능하게 한다.
시간복잡도 O(nlogn)
'''
def solution(numbers):
    numbers.sort(key = lambda x:list(str(x)) * 3, reverse = True)
    answer = str(int(''.join(list(map(str, numbers))))) 
    #11번 테스트 케이스 [0,0,0,0] -> "0", str(int()) "0000" -> "0"으로 바꾸기 위해사용 
    return answer